packages/workspace/src/command-line/affected.ts

```diff
@@ -45,7 +45,12 @@ export async function affected(
     );
   }
   const projects = parsedArgs.all ? projectGraph.nodes : affectedGraph.nodes;
-  const env = readEnvironment(nxArgs.target, projects);
+  const projectsNotExcluded = Object.keys(projects)
+    .filter((key) => !parsedArgs.exclude.includes(key))
+    .reduce((_p, key) => {
+      return projects[key];
+    }, {});
+  const env = readEnvironment(nxArgs.target, projectsNotExcluded);
   const affectedProjects = Object.values(projects)
```

Comment:
Could you pass projectsNotExcluded here?

Reply:
i had a bad reduce, corrected. Sorry for that !

i also pass `projectsNotExcluded` in commit 5c80987027879222a0adc5fd6c41a86f2d91aa3d